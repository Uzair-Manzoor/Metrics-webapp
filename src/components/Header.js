const Header = () => (
  <header className="header flex">
    <h1>Weather and Pollution Metrics for Canada</h1>
  </header>
);

export default Header;
