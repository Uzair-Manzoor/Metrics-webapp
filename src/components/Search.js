import React, { useState } from 'react';
import PropTypes from 'prop-types';

const Search = ({ onSearch }) => {
  const [query, setQuery] = useState('');

  const handleInputChange = (e) => {
    const newQuery = e.target.value;
    setQuery(newQuery);
    onSearch(newQuery);
  };

  return (
    <div className="search-container">
      <input
        type="text"
        placeholder="Search..."
        value={query}
        onChange={handleInputChange}
      />
    </div>
  );
};

Search.propTypes = {
  onSearch: PropTypes.func.isRequired,
};

export default Search;
