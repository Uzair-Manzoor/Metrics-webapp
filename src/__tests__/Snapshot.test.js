import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../components/Header';
import Search from '../components/Search';

describe('Snapshot Tests', () => {
  it('renders Header correctly', () => {
    const headerComponent = renderer.create(<Header />).toJSON();
    expect(headerComponent).toMatchSnapshot();
  });

  it('renders Search correctly', () => {
    const searchComponent = renderer.create(<Search onSearch={() => {}} />).toJSON();
    expect(searchComponent).toMatchSnapshot();
  });
});
