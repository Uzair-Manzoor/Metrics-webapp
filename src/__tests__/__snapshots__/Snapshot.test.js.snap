// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests renders Header correctly 1`] = `
<header>
  <h1>
    Pollution Metrics for Canada
  </h1>
</header>
`;

exports[`Snapshot Tests renders Search correctly 1`] = `
<div
  className="search-container"
>
  <input
    data-testid="search-text"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
</div>
`;
